### PR TITLE
EOS-19354: TMPL_POOL_TYPE is not used in the templates

### PR DESCRIPTION
**Pull request by knekrasov**
JIRA issue: [EOS-19354](https://jts.seagate.com/browse/EOS-19354)

Solution: modify the template files so that TMPL_POOL_TYPE placeholder
is used instead of direct value (sns, dix).

### CODE REVIEW

**Review by vaibhavparatwar — COMMENTED** [review]
@knekrasov can you please add these two keys as well as part of this PR? - storage_set_id and cvg_count - please refer PR #1731 ? in #1731 we added only in hare.config.conf.tmpl.1-node file

**Comment by knekrasov** [comment]
@vaibhavparatwar I addressed the discrepancies in our template files and implemented the corresponding auto test. Please check.

File: provisioning/miniprov/test/test_templates.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import logging
```
Comment (mssawant) [inline provisioning/miniprov/test/test_templates.py:19]: Kindly fix the codacy issue.

File: provisioning/miniprov/test/test_templates.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import os
+import re
+import tempfile
+from typing import Callable, Dict, List
```
Comment (mssawant) [inline provisioning/miniprov/test/test_templates.py:23]: Kindly fix the codacy issue.

File: provisioning/miniprov/test/test_templates.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        'TMPL_S3SERVER_INSTANCES_COUNT': '4',
+        'TMPL_SERVER_NODE_NAME': 'srvnode-1',
+        'TMPL_SPARE_UNITS_COUNT': '0',
+        'TMPL_STORAGESET_COUNT': '1',
+        'TMPL_STORAGESET_NAME': 'storage-set',
+        'TMPL_STORAGE_SET_ID': 'storage-set',
+    }
+
+
+@pytest.mark.parametrize('tmpl_file', template_files())
```
Comment (mssawant) [inline provisioning/miniprov/test/test_templates.py:101]: Good idea!

**Review by mssawant — APPROVED** [review]
Looks good, just need to address few codacy reported issues.

**Review by vaibhavparatwar — APPROVED** [review]

File: provisioning/miniprov/test/test_templates.py (hunk 2 above)
Comment (knekrasov) [reply provisioning/miniprov/test/test_templates.py]: Addressed, thanks!

File: provisioning/miniprov/test/test_templates.py (hunk 1 above)
Comment (knekrasov) [reply provisioning/miniprov/test/test_templates.py]: Done

**Comment by suvratjoshi** [comment]
@knekrasov 
I think it would be nice to add 1 dix pools as well to all the sample files.
Currently we just have sns.
Thanks!

**Comment by knekrasov** [comment]
I think it would be nice to add 1 dix pool as well to all the sample files.
Currently we just have sns.

@suvratjoshi I guess this gets too far away of the PR title :-)

**Review by suvratjoshi — APPROVED** [review]

**Pull request merged**